util/dvsim/Launcher.py

```diff
@@ -241,6 +245,28 @@ def _find_patterns(patterns, line):
                     return pattern
             return None
 
+        def _get_runtimes(lines, tool):
+            """Helper function that assigns job_runtime and assigns simulated_time if the deploy
+            target is `run`."""
+            try:
+                self.job_runtime = get_job_runtime(lines, tool)
+            except NotImplementedError as e:
+                #TODO: use dvsim default runtime.
+                self.job_runtime = 0
+            except SyntaxError as e:
+                #TODO: use dvsim default runtime.
+                self.job_runtime = 0
+                log.debug("Parsing job runtime has syntax error with " + tool)
+
+            if self.deploy.target == "run":
+                self.deploy.job_runtime = self.job_runtime
+                try:
+                    self.deploy.simulated_time = get_simulated_time(lines, tool)
+                except NotImplementedError as e:
+                    log.debug("Parsing simulated time with unsupported tool: " + tool)
+                except SyntaxError as e:
+                    log.debug("Parsing job simulated time has syntax error with " + tool)
```

Comment:
Since `simulated_time` is only maintained by `RunTest`, this logic belongs to `RunTest::post_finish`, which is nice because you do not need to check the `target` value.

Reply:
@sriyerg Sorry i am slightly confused here ;) Are you suggesting to move this logic to `RunTest::post_finish`? I thought the reason we move the `get_runtime` here is to avoid opening the file too many times?
Thanks again for your review and patience!